language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/semantictokens/SemanticTokensVisitor.java

```diff
@@ -580,33 +579,30 @@ private void processSymbols(Node node, LinePosition startLine) {
                 break;
         }
 
-        Optional<String> path =
-                this.semanticTokensContext.workspace().relativePath(this.semanticTokensContext.filePath());
-        if (path.isEmpty()) {
-            return;
-        }
         // Add the symbol's semantic token if it is in the same file
-        if (declarationType != -1) {
-            if (symbolLineRange.filePath().equals(path.get()) && symbol.get().getModule().isPresent() &&
-                    symbol.get().getModule().get().getName().isPresent() && this.semanticTokensContext
-                    .currentModule().isPresent() && symbol.get().getModule().get().getName().get()
-                    .equals(this.semanticTokensContext.currentModule().get().moduleId().moduleName())) {
+        if (declarationType != -1 && symbolLineRange.filePath().equals(this.semanticTokensContext.currentDocument()
+                .get().name()) && symbol.get().getModule().isPresent() && symbol.get().getModule().get().getName()
+                .isPresent() && this.semanticTokensContext.currentModule().isPresent() && symbol.get().getModule()
+                .get().getName().get().equals(this.semanticTokensContext.currentModule().get().moduleId()
+                        .moduleName())) {
                 SemanticToken semanticToken = new SemanticToken(linePosition.line(), linePosition.offset());
-                if (!semanticTokens.contains(semanticToken)) {
-                    semanticToken.setProperties(node.textRange().length(), declarationType,
-                            declarationModifiers == -1 ? 0 : declarationModifiers);
-                    semanticTokens.add(semanticToken);
-                }
+            if (!semanticTokens.contains(semanticToken)) {
+                semanticToken.setProperties(node.textRange().length(), declarationType, declarationModifiers == -1 ?
+                        0 : declarationModifiers);
+                semanticTokens.add(semanticToken);
             }
         }
 
         // Add symbol's references if they reside in the same file
         if (referenceType != -1) {
             final int type = referenceType;
             final int modifiers = referenceModifiers == -1 ? 0 : referenceModifiers;
-            semanticModel.get().references(symbol.get(), false).stream().filter(location ->
-                    location != null && location.lineRange().filePath().equals(path.get()))
-                    .forEach(location -> {
+
+            List<Location> locations = semanticModel.get().references(symbol.get(),
+                    this.semanticTokensContext.currentDocument().get(), false);
+            locations.stream().filter(location ->
+                    location != null && location.lineRange().filePath()
```

Comment:
can location be null here? the symbol location shouldn't be null. If it is null, then we have to fix. let's avoid the null check here

Reply:
Earlier there was a test failure with a null location. Seems to have been fixed now in master. Will update.